libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -407,22 +404,20 @@ bool IRGeneratorForStatements::visit(Assignment const& _assignment)
 		assignmentOperator :
 		TokenTraits::AssignmentToBinaryOp(assignmentOperator);
 
-	Type const* rightIntermediateType =
-		TokenTraits::isShiftOp(binaryOperator) ?
-		type(_assignment.rightHandSide()).mobileType() :
-		type(_assignment.rightHandSide()).closestTemporaryType(
-			&type(_assignment.leftHandSide())
-		);
-	solAssert(rightIntermediateType, "");
-	IRVariable value = convert(_assignment.rightHandSide(), *rightIntermediateType);
+	solAssert(!TokenTraits::isShiftOp(binaryOperator) || type(_assignment.rightHandSide()).mobileType(), "");
```

Comment:
Can you write this as `if(isShiftOp) solAssert(...)`?

Reply:
Will do like that